src/platform/Linux/bluez/Helper.cpp

```diff
@@ -417,6 +426,8 @@ static void Bluez_gatt_characteristic1_complete_acquire_write_with_fd(GDBusMetho
 
 static gboolean bluezCharacteristicDestroyFD(GIOChannel * aChannel, GIOCondition aCond, gpointer apClosure)
 {
+    BluezConnection * conn  = static_cast<BluezConnection *>(apClosure);
+    conn->mC2Channel.mWatch = 0;
```

Comment:
Looks like a similar comment is useful here: why are we setting mWatch to 0?

Reply:
Added the same commend as for `conn->mC1Channel.mWatch = 0`.